workflows/sampletracking.nf

```diff
@@ -40,12 +42,34 @@ workflow SAMPLETRACKING {
     def ch_multiqc_files = Channel.empty()
     def ch_pool_multiqc_files = Channel.empty()
 
+    def ch_bams_with_index = ch_samplesheet.filter { meta, sample_bam, sample_bam_index, snp_fastq, snp_bam, snp_bam_index ->
+        sample_bam_index != null
+    }
+    def ch_bams_no_index = ch_samplesheet.filter { meta, sample_bam, sample_bam_index, snp_fastq, snp_bam, snp_bam_index ->
+        sample_bam_index == null
+    }
```

Comment:
This is one good way to handle this, but IMO [`.branch`](https://www.nextflow.io/docs/latest/reference/operator.html#branch) is more readable and less verbose than `.filter`